lib/rubocop/cop/rspec/capybara/negation_matcher.rb

```diff
@@ -0,0 +1,104 @@
+        #
+        #   # good
+        #   expect(page).to have_no_selector
+        #   expect(page).to have_no_css('a')
+        #
+        # @example EnforcedStyle: not_to
+        #   # bad
+        #   expect(page).to have_no_selector
+        #   expect(page).to have_no_css('a')
+        #
+        #   # good
+        #   expect(page).not_to have_selector
+        #   expect(page).not_to have_css('a')
+        #
+        class NegationMatcher < Base
+          extend AutoCorrector
+          include ConfigurableEnforcedStyle
+
+          MSG = 'Use `%<good>s` instead of `%<bad>s`.'
+          CAPYBARA_MATCHERS = %w[
+            selector css xpath text title current_path link button
+            field checked_field unchecked_field select table
+            sibling ancestor
+          ].freeze
+          POSITIVE_MATCHERS = CAPYBARA_MATCHERS.map do |element|
+            "have_#{element}".to_sym
+          end.freeze
+          NEGATIVE_MATCHERS = CAPYBARA_MATCHERS.map do |element|
+            "have_no_#{element}".to_sym
+          end.freeze
+          RESTRICT_ON_SEND = (POSITIVE_MATCHERS + NEGATIVE_MATCHERS).freeze
+
+          # @!method not_to?(node)
+          def_node_matcher :not_to?, <<~PATTERN
+            (send ... :not_to
+              (send nil? {#{POSITIVE_MATCHERS.map(&:inspect).join(' ')}} ...))
+          PATTERN
+
+          # @!method have_no?(node)
+          def_node_matcher :have_no?, <<~PATTERN
+            (send ... :to
+              (send nil? {#{NEGATIVE_MATCHERS.map(&:inspect).join(' ')}} ...))
+          PATTERN
+
+          def on_send(node)
+            return unless offense?(node.parent)
+
+            add_offense(offense_range(node),
+                        message: message(node)) do |corrector|
+              corrector.replace(node.parent.loc.selector, replaced_target)
+              corrector.replace(node.loc.selector, replaced_matcher(node))
+            end
+          end
+
+          private
+
+          def offense?(node)
+            (style == :have_no && not_to?(node)) ||
+              (style == :not_to && have_no?(node))
+          end
+
+          def offense_range(node)
+            node.parent.loc.selector.with(end_pos: node.loc.selector.end_pos)
+          end
+
+          def message(node)
+            format(MSG,
+                   good: format_message(node.parent.loc.selector.source,
+                                        node.loc.selector.source),
+                   bad: format_message(replaced_target, replaced_matcher(node)))
+          end
+
+          def format_message(target, matcher)
+            "expect(...).#{target} #{matcher}"
+          end
+
+          def replaced_target
+            style == :have_no ? 'to' : 'not_to'
+          end
+
+          def replaced_matcher(node)
+            if style == :have_no
+              node.loc.selector.source.gsub('have_', 'have_no_')
+            else
+              node.loc.selector.source.gsub('_no', '')
```

Comment:
Maybe `.gsub('have_no_', 'have_')` to avoid some edge cases?

Reply:
Indeed it is. I updated it.